projects/knora-ui/src/lib/viewer/values/link-value/link-value.component.ts

```diff
@@ -0,0 +1,169 @@
+import {Component, Inject, Input, OnChanges, OnDestroy, OnInit, SimpleChanges} from '@angular/core';
+import {BaseValueComponent} from '../base-value.component';
+import {CreateLinkValue, ReadLinkValue, ReadResource, UpdateLinkValue, KnoraApiConnection} from '@knora/api';
+import {Subscription} from 'rxjs';
+import {AbstractControl, FormBuilder, FormControl, FormGroup, ValidatorFn} from '@angular/forms';
+import {KnoraApiConnectionToken} from '../../../core';
+@Component({
+  selector: 'kui-link-value',
+  templateUrl: './link-value.component.html',
+  styleUrls: ['./link-value.component.scss']
+})
+export class LinkValueComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() displayValue?: ReadLinkValue;
+  @Input() parentResource: ReadResource;
+  @Input() propType: string;
+  resources: ReadResource[];
+  restrictToResourceClass: string;
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+  form: FormGroup;
+
+  valueChangesSubscription: Subscription;
+  // label cannot contain logical operations of lucene index
+  customValidators = [];
+
+  constructor(@Inject(FormBuilder) private fb: FormBuilder, @Inject(KnoraApiConnectionToken) private knoraApiConnection: KnoraApiConnection) {
+    super();
+  }
+
+  /**
+   * Displays a selected resource using its label.
+   *
+   * @param resource the resource to be displayed (or no selection yet).
+   * @returns
+   */
+  displayResource(resource: ReadResource | null): string {
+
+    // null is the initial value (no selection yet)
+    if (resource !== null) {
+      return resource.label;
+    }
+  }
+
+  /**
+   * Search for resources whose labels contain the given search term, restricting to to the given properties object constraint.
+   * this is to be used for update and new linked resources
+   *
+   * @param label to be searched
+   */
+  searchByLabel(searchTerm: string): ReadResource[] {
+    const linkType = this.parentResource.getLinkPropertyIriFromLinkValuePropertyIri(this.propType);
```

Comment:
Could this be evaluated only once?

Reply:
hmm, yes. I can move it to ngOnInit

Reply:
I think that would do it, since the object type of the property does not change (unless of course there are changes in the ontology). 

Could you put that logic in a method and write a test for it?

Reply:
I think with https://github.com/dasch-swiss/knora-api-js-lib/pull/168 having been merged, this check is not necessary anymore.

Reply:
> hmm, yes. I can move it to ngOnInit

Good idea!

Reply:
I have already moved the call to this function into ngOnInit.

Reply:
but

```const linkType = this.parentResource.getLinkPropertyIriFromLinkValuePropertyIri(this.propType);
this.restrictToResourceClass = this.parentResource.entityInfo.properties[linkType].objectType;```

is still in `searchByLabel `. Sorry, I am confused now.

Reply:
what I meant is: instead of re-evaluating the link property's object type for each search, it could be just done once.

Reply:
Ah, I see. The confusion was due to the previous discussion where we discussed the search for label should be in OnInit not in resetFormControl. I just moved the evaluation of the this.restrictResourceClass to the ngOnInit.